src/main/java/io/cryostat/agent/remote/RecordingsContext.java

```diff
@@ -106,12 +109,23 @@ public void handle(HttpExchange exchange) throws IOException {
                     }
                     break;
                 case "POST":
-                    handleStart(exchange);
+                    handleStartRecordingOrSnapshot(exchange);
                     break;
                 case "PATCH":
                     id = extractId(exchange);
-                    if (id >= 0) {
-                        handleStop(exchange, id);
+                    String request = requestStopOrUpdate(exchange);
+                    if (request == "STOPPED") {
```

Comment:
Careful - in Java, Strings should be compared like `x.equals("A")`, not `x == "A"`, since the `==` only checks reference equality.